compute/disks/disks_test.go

```diff
@@ -0,0 +1,270 @@
+	if err != nil {
+		return err
+	}
+
+	return op.Wait(ctx)
+}
+
+func getInstance(
+	ctx context.Context,
+	projectID, zone, instanceName string,
+) (*computepb.Instance, error) {
+	instancesClient, err := compute.NewInstancesRESTClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer instancesClient.Close()
+	reqInstance := &computepb.GetInstanceRequest{
+		Project:  projectID,
+		Zone:     zone,
+		Instance: instanceName,
+	}
+
+	return instancesClient.Get(ctx, reqInstance)
+}
+
+func createDiskSnapshot(ctx context.Context, projectId, zone, diskName, snapshotName string) error {
+	disksClient, err := compute.NewDisksRESTClient(ctx)
+	if err != nil {
+		return err
+	}
+	defer disksClient.Close()
+	req := &computepb.CreateSnapshotDiskRequest{
+		Project: projectId,
+		Zone:    zone,
+		Disk:    diskName,
+		SnapshotResource: &computepb.Snapshot{
+			Name: proto.String(snapshotName),
+		},
+	}
+
+	op, err := disksClient.CreateSnapshot(ctx, req)
+	if err != nil {
+		return err
+	}
+
+	return op.Wait(ctx)
+}
+
+func deleteDiskSnapshot(ctx context.Context, projectId, snapshotName string) error {
+	snapshotsClient, err := compute.NewSnapshotsRESTClient(ctx)
+	if err != nil {
+		return err
+	}
+	defer snapshotsClient.Close()
+	req := &computepb.DeleteSnapshotRequest{
+		Project:  projectId,
+		Snapshot: snapshotName,
+	}
+
+	op, err := snapshotsClient.Delete(ctx, req)
+	if err != nil {
+		return err
+	}
+
+	return op.Wait(ctx)
+}
+
+func deleteInstance(ctx context.Context, projectId, zone, instanceName string) error {
+	instancesClient, err := compute.NewInstancesRESTClient(ctx)
+	if err != nil {
+		return err
+	}
+	defer instancesClient.Close()
+	req := &computepb.DeleteInstanceRequest{
+		Project:  projectId,
+		Zone:     zone,
+		Instance: instanceName,
+	}
+
+	op, err := instancesClient.Delete(ctx, req)
+	if err != nil {
+		return err
+	}
+
+	return op.Wait(ctx)
+}
+
+func TestComputeDisksSnippets(t *testing.T) {
+	ctx := context.Background()
+	var r *rand.Rand = rand.New(
+		rand.NewSource(time.Now().UnixNano()))
+	tc := testutil.SystemTest(t)
+	zone := "europe-central2-b"
+	instanceName := fmt.Sprintf("test-instance-%v-%v", time.Now().Format("01-02-2006"), r.Int())
+	diskName := fmt.Sprintf("test-disk-%v-%v", time.Now().Format("01-02-2006"), r.Int())
+	diskName2 := fmt.Sprintf("test-disk-%v-%v", time.Now().Format("01-02-2006"), r.Int())
+	snapshotName := fmt.Sprintf("test-snapshot-%v-%v", time.Now().Format("01-02-2006"), r.Int())
+	sourceImage := "projects/debian-cloud/global/images/family/debian-11"
+	diskType := fmt.Sprintf("zones/%s/diskTypes/pd-ssd", zone)
+	want := "Disk created"
+
+	instancesClient, err := compute.NewInstancesRESTClient(ctx)
+	if err != nil {
+		t.Fatalf("NewInstancesRESTClient: %v", err)
+	}
+
+	defer instancesClient.Close()
+
+	buf := &bytes.Buffer{}
+
+	err = createDisk(ctx, tc.ProjectID, zone, diskName, sourceImage)
+	if err != nil {
+		t.Errorf("createDisk got err: %v", err)
```

Comment:
Many of these should be a t.Fatalf. Errorf allows the test to keep running after failure, but that is not what is desired here.